engine/crates/runtime/src/fetch.rs

```diff
@@ -60,6 +61,39 @@ impl Fetcher {
             inner: Arc::new(fetcher),
         }
     }
+
+    #[allow(unused_variables)]
+    pub async fn post<SleepFut>(
+        &self,
+        request: &FetchRequest<'_>,
+        sleep: impl Fn(Duration) -> SleepFut,
+    ) -> FetchResult<FetchResponse>
+    where
+        SleepFut: std::future::Future<Output = ()>,
+    {
```

Comment:
This is where the retries happen. We're missing backoff, preferably with jitter.